src/libraries/System.Private.CoreLib/src/System/Text/EncodingTable.cs

```diff
@@ -105,18 +106,23 @@ private static int InternalGetCodePageFromName(string name)
         // Return a list of all EncodingInfo objects describing all of our encodings
         internal static EncodingInfo[] GetEncodings()
```

Comment:
>GetEncodings [](start = 39, length = 12)

After my change enabling the encoding enumeration through the providers, I am wondering should we allow UTF7 provided by the provider? I would say not allowing it except if the config switch is turned on.

Reply:
If somebody went out of their way to create a custom provider that implements UTF-7, why would we disable it? They've already opted-in by virtue of registering the provider.

For context, when I spoke offline with Shawn the recommendation was that _all_ non-Unicode encodings should be disabled by default. This means that arbitrary code page encodings would be disabled by default as well. And it turns out this is already the case, since unless you register the code page provider, the .NET runtime doesn't know anything about how to resolve those encodings.

Reply:
> If somebody went out of their way to create a custom provider that implements UTF-7, why would we disable it? They've already opted-in by virtue of registering the provide

I believe enabling or disabling the UTF-7 decision should be taken by the app (through the config switch) and not any other place. Imagine a library using a provider that has UTF-7 (even the library not caring about this encoding) and then the app using such a library. That means the library took the decision on behalf of the app which I think is not right. Another point is consistency and easy to understand the behavior regardless of there are providers or not.

> when I spoke offline with Shawn the recommendation was that all non-Unicode encodings should be disabled by default.

This is proven impossible as we saw a lot of scenarios that require non-Unicode encoding. like networking protocols using ISO encodings for instance. I know Shawn passionate for Unicode which I share that too but the reality is really different.

Reply:
Of course. I mentally appended "plus ASCII and Latin-1" to what he said, even if those exact words never came from his mouth. 😉

Reply:
:-)

Also, .NET has to be compliant and support specific encodings (e.g. GB13080 for instance). disabling these by default can create unwanted issues :-)